core/src/main/java/org/apache/accumulo/core/metadata/ValidationUtil.java

```diff
@@ -52,8 +56,13 @@ public static Path validate(Path path) {
   }
 
   public static void validateRFileName(String fileName) {
-    Objects.requireNonNull(fileName);
-    // TODO: In 3.0.0 validate that filename starts with FilePrefix
+    Preconditions.checkArgument(!StringUtils.isEmpty(fileName),
+        "Provided filename is null or empty.");
+
+    // This will make sure the prefix is valid and throw an IllegalArgumentException if the prefix
+    // does not match a known value
+    var unused = FilePrefix.fromPrefix(fileName.substring(0, 1));
```

Comment:
Naming the variable "unused" doesn't actually prevent some tools from reporting that it is unused. Could do some kind of assertion check here (`if (FilePrefix.fromPrefix(...) == null) { throw .... ; }`). Depending on what it returns, this check could probably be combined with the subsequent `if` statement that verifies it the file name `.endsWith` the expected stuff.

Or just add a trivial `void` method in `FilePrefix` called `validatePrefix` that works similarly to `fromPrefix`, but doesn't have an expected return statement (and maybe does the substring for you).

Reply:
I had put that in there as I've seen (I think) SpotBugs complain in the past but can definitely clean that up when I get some more time to work on this.